### PR TITLE
Changes to log messages into individual log file for each It class

### PR DESCRIPTION
**Pull request by vanajamukkara**
- log messages into individual log file for each It class, same for sequential or parallel run.
- fix port conflict issues with nginx when tests run in parallel, made by Xian.

This PR is mainly for logging messages. There are failures when tests ran in parallel which needs to be looked into. There will be a separate PR to make the tests work when run in parallel.

Jenkins Results -- intermittent failures not related to the changes in the PR
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/529/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-model-in-image-tests-1/48/

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 1]
```diff
@@ -34,7 +34,9 @@
 import oracle.weblogic.kubernetes.annotations.Namespaces;
 import oracle.weblogic.kubernetes.annotations.tags.MustNotRunInParallel;
 import oracle.weblogic.kubernetes.annotations.tags.Slow;
-import oracle.weblogic.kubernetes.extensions.LoggedTest;
+//import oracle.weblogic.kubernetes.extensions.LoggedTest;
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:37]: remove commented out import

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiConfigMap.java [hunk 2]
```diff
@@ -107,6 +109,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiConfigMap.java:112]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiConfigMapOverride.java [hunk 3]
```diff
@@ -118,6 +120,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiConfigMapOverride.java:123]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java [hunk 4]
```diff
@@ -132,6 +135,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(3) List<String> namespaces) {
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiDomain.java:138]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiMultiModel.java [hunk 5]
```diff
@@ -119,7 +122,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(2) List<String> namespaces) {
-
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiMultiModel.java:125]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiSample.java [hunk 6]
```diff
@@ -89,7 +91,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(5) List<String> namespaces) {
-
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiSample.java:94]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java [hunk 7]
```diff
@@ -184,7 +187,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(6) List<String> namespaces) {
-
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMonitoringExporter.java:190]: same here

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorTwoDomains.java [hunk 8]
```diff
@@ -129,7 +131,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(4) List<String> namespaces) {
-
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOperatorTwoDomains.java:134]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java [hunk 9]
```diff
@@ -80,7 +82,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(2) List<String> namespaces) {
-
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItPodsRestart.java:85]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSessionMigration.java [hunk 10]
```diff
@@ -100,6 +102,7 @@
    */
   @BeforeAll
   public static void init(@Namespaces(2) List<String> namespaces) {
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSessionMigration.java:105]: same here

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleValidation.java [hunk 11]
```diff
@@ -61,27 +62,31 @@
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
+//import oracle.weblogic.kubernetes.extensions.LoggedTest;
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleValidation.java:65]: remove commented out import

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleValidation.java [hunk 12]
```diff
@@ -61,27 +62,31 @@
...
+
   /**
    * Setup for test suite. Creates service account, persistent volume
    * and persistent volume claim.
    * @param namespaces injected by JUnit extension
    */
   @BeforeAll
   public void setup(@Namespaces(2) List<String> namespaces) {
-
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleValidation.java:89]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleValidation.java [hunk 13]
```diff
@@ -232,7 +237,7 @@ public void testInstallingOperator() {
   /**
    * Create domain.
    */
-  @Test
+  //@Test
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSimpleValidation.java:240]: either enable the test or delete this test method

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java [hunk 14]
```diff
@@ -98,7 +100,7 @@
    */
   @BeforeAll
   public static void initAll(@Namespaces(3) List<String> namespaces) {
-
+    logger = ThreadSafeLogger.getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItUsabilityOperatorHelmChart.java:103]: you can static import ThreadSafeLogger.getLogger then just call getLogger() here like in other It* classes.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java [hunk 15]
```diff
@@ -105,7 +107,7 @@ public static boolean shutdown(String domainUid, String namespace) {
    * @return true if patching domain resource succeeded, false otherwise
    */
   public static boolean restart(String domainUid, String namespace) {
-
+    LoggingFacade logger = getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:110]: can we move the logger definition to the top so we don't need to make two logger definition here.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java [hunk 16]
```diff
@@ -173,7 +175,7 @@ public static boolean patchDomainCustomResource(String domainUid, String namespa
    */
   public static boolean patchDomainResourceWithNewIntrospectVersion(
       String domainUid, String namespace) throws ApiException {
-
+    LoggingFacade logger = getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:178]: same here

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java [hunk 17]
```diff
@@ -268,7 +270,7 @@ public static boolean scaleClusterWithRestApi(String domainUid,
                                                 int externalRestHttpsPort,
                                                 String opNamespace,
                                                 String opServiceAccount) {
-
+    LoggingFacade logger = getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:273]: same here

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java [hunk 18]
```diff
@@ -354,7 +356,7 @@ public static boolean scaleClusterWithWLDF(String clusterName,
                                              String myWebAppName,
                                              String curlCommand)
       throws ApiException, IOException, InterruptedException {
-
+    LoggingFacade logger = getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:359]: same here

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java [hunk 19]
```diff
@@ -215,7 +217,7 @@ public static boolean patchDomainResourceWithNewIntrospectVersion(
    */
   public static boolean scaleCluster(String domainUid, String namespace, String clusterName, int numOfServers)
       throws ApiException {
-
+    LoggingFacade logger = getLogger();
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:220]: same here

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java (hunk 15 above)
Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:110]: do you mean make it class level?

Comment (xiancao) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:110]: yes

Comment (vanajamukkara) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/Domain.java:110]: as explained in slack, can't use the class level variable here, it has to be local variable to get the logger from current thread

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/extensions/ImageBuilders.java [hunk 20]
```diff
@@ -69,13 +71,16 @@
...
-import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
 import static oracle.weblogic.kubernetes.utils.FileUtils.checkDirectory;
 import static oracle.weblogic.kubernetes.utils.FileUtils.cleanupDirectory;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
 import static org.junit.jupiter.api.Assertions.assertFalse;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 import static org.junit.jupiter.api.extension.ExtensionContext.Namespace.GLOBAL;
 
+//import oracle.weblogic.kubernetes.logging.LoggingFacade;
+//import static oracle.weblogic.kubernetes.extensions.LoggedTest.logger;
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/extensions/ImageBuilders.java:82]: remove commented out imports

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/extensions/LoggingExtension.java [hunk 21]
```diff
@@ -0,0 +1,25 @@
...
+
+    ThreadSafeLogger.globalLogger.info("Initializing logger in postProcessTestInstance for {0}",
+        testInstance.getClass().getSimpleName());
+
+    // initialize logger for each test
+    ThreadSafeLogger.init(testInstance.getClass().getSimpleName());
+
+  }
+
+
```
Comment (xiancao) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/extensions/LoggingExtension.java:24]: remove empty line here

**Pull request merged**